model/calendar.js

```diff
@@ -28,6 +31,85 @@ class Calendar extends Google {
     return this.filterResults(events);
   }
 
+  // private
+  getEventStartDate(event) {
+    if (!event.start) {
+      return null;
+    }
+
+    if (event.start.dateTime) {
+      return moment(event.start.dateTime); // TODO add proper parsing
+    }
+
+    if (event.start.date) {
+      // this is an all-day event
+      return null;
+    }
+
+    throw new Error('Unexpected start date pattern: ' + JSON.stringify(event.start));
+  }
+
+  // private
+  getEventEndDate(event) {
+    if (!event.end) {
+      return null;
+    }
+
+    if (event.end.dateTime) {
+      return moment(event.end.dateTime); // TODO add proper parsing
+    }
+
+    if (event.end.date) {
+      // this is an all day event
+      return null;
+    }
+
+    throw new Error('Unexpected end date pattern: ' + JSON.stringify(event.end));
+  }
+
+  // private
+  getEventDays(event) {
+    const days = [];
+    const current = this.getEventStartDate(event);
+    const end = this.getEventEndDate(event);
+
+    if (!(current && end)) {
+      return [];
+    }
+
+    while (true) {
+      days.push(current.format(YYYYMMDD));
+
+      if (current.format(YYYYMMDD) === end.format(YYYYMMDD)) {
+        break;
+      }
+
+      current.add(1, 'day');
+    }
+
+    return days;
+  }
+
+  // private
+  getWorkingHoursForDay(day) {
+    const earliestEventOfTheDay = day.sort((a, b) => (a.started - b.started))[0];
+    const lastEventOfTheDay = day.sort((a, b) => (b.ended - a.ended))[0];
```

Comment:
@vadimburlakin I think you changed the logic, no? You should explain in the description it better.
For example, I would prefer much more to sum the events than count the earliest and the latest, because I can have huge gaps in the middle and it doesn't work well when working from another timezone.